File: backend/app/api/routes/analytics.py
from fastapi import APIRouter

from backend.app.services import alert_service, camera_service

router = APIRouter(prefix="/analytics", tags=["analytics"])


@router.get("/summary")
def summary() -> dict:
    cameras = camera_service.list_cameras()
    alerts = alert_service.list_alerts()
    by_severity = {"high": 0, "medium": 0, "low": 0}
    for alert in alerts:
        by_severity[alert.severity] = by_severity.get(alert.severity, 0) + 1
    return {
        "cameras_online": sum(1 for camera in cameras if camera.status == "online"),
        "cameras_total": len(cameras),
        "alerts_open": sum(1 for alert in alerts if alert.status == "open"),
        "alerts_by_severity": by_severity,
    }

File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  cameras_online: number;
  cameras_total: number;
  alerts_open: number;
  alerts_by_severity: {
    high: number;
    medium: number;
    low: number;
    [key: string]: number;
  };
}

export interface SectorThreatData {
  sector: string;
  alerts: number;
  events: number;
}

export interface TimelineDataPoint {
  time: string;
  intrusions: number;
  loitering: number;
  groups: number;
}
